updating Knocki docs (#34654)
home-assistant/home-assistant.github.io: source/_integrations/knocki.markdown

ha_integration_type: device
---

The **Knocki** {% term integration %} allows you to trigger your favorite automations simply by tapping custom patterns (such as triple taps) on ordinary surfaces.

## Prerequisites

- You must have a [Knocki device](http://knocki.com).

> - Completed the setup process in the knocki mobile app (steps 1-7)

Reviewer: ```suggestion
- Completed the setup process in the **Knocki** mobile app, as described in steps 1-7 below.
```

1. Open the Knocki Mobile App. It’s a free download on [iOS](https://apps.apple.com/us/app/knocki/id1238395440) and [Android](https://play.google.com/store/apps/details?id=com.knocki.mobileapp).
2. Using the app, log-in or create a new Knocki account.
3. From the app’s home screen, select a Knocki or [add a Knocki](https://support.knocki.com/hc/en-us/articles/12769368448659-How-Do-Add-a-Knocki-to-My-Account).
4. On the app’s device screen, select a gesture or [add a gesture](https://support.knocki.com/hc/en-us/articles/360013333634-Creating-a-Gesture).
5. Next, select the [add a task](https://support.knocki.com/hc/en-us/articles/12920956118291-How-do-I-Add-Tasks-to-a-Knocki) option.
6. Scroll down the task menu and select Home Assistant.
7. Follow the in-app instructions to add the Home Assistant task.

> {% include integrations/config_flow.md %}

Reviewer: ```suggestion
{% include integrations/config_flow.md %}

## Automation example

1. Go to {% my automations title="**Settings** > **Automations & scenes**" %} and in the lower right corner, select  **Create Automation**.
2. Select **Create new automation**.
3. Select **Add Trigger**, select **Entity** as the trigger type.  
   - Search for the name of the task (as you named it in the Knocki app). Select the entity with that name appended to it.  
   - Select **State** as the type of entity change.
4. Select **Add action** and choose whatever action(s) you’d like the automation to trigger.
5. Finally, test the action by tapping the associated gesture pattern. 
  - **Troubleshooting**: If the new automation isn’t triggering successfully, select the **reload** option from the devices page for Knocki on Home Assistant. It should then start working.
```
Readded automation steps in separate section. Could you check if they are still correct?

Reviewer 2: I think it looks correct

Author: @c0ffeeca7 Looks good to me. I pushed the same changes but added `Troubleshooting` as a new section.

ha_integration_type: device
---

The **Knocki** {% term integration %} allows you to trigger your favorite automations simply by tapping custom patterns (such as triple taps) on ordinary surfaces.

> ## Requirements

Reviewer: ```suggestion
## Prerequisites
```

> Before you begin the setup process, you must have a [Knocki device](http://knocki.com).

Reviewer: ```suggestion
- You must have a [Knocki device](http://knocki.com).
```

> ## Setup Instructions
> 

Reviewer: ```suggestion
```

> Start the setup process in the knocki mobile app (steps 1-7), and complete the process in Home Assistant (steps 8-13).

Reviewer: ```suggestion
- Completed the setup process in the knocki mobile app (steps 1-7)
```

1. Open the Knocki Mobile App. It’s a free download on [iOS](https://apps.apple.com/us/app/knocki/id1238395440) and [Android](https://play.google.com/store/apps/details?id=com.knocki.mobileapp).
2. Using the app, log-in or create a new Knocki account.
3. From the app’s home screen, select a Knocki or [add a Knocki](https://support.knocki.com/hc/en-us/articles/12769368448659-How-Do-Add-a-Knocki-to-My-Account).
4. On the app’s device screen, select a gesture or [add a gesture](https://support.knocki.com/hc/en-us/articles/360013333634-Creating-a-Gesture).
5. Next, select the [add a task](https://support.knocki.com/hc/en-us/articles/12920956118291-How-do-I-Add-Tasks-to-a-Knocki) option.
6. Scroll down the task menu and select Home Assistant.
7. Follow the in-app instructions to add the Home Assistant task.

> 8. In Home Assistant, add the Knocki Integration by clicking the blue "Add Integration" button.
> 9. You should see entities for each Knocki Task that you already created in the Knocki app, and the corresponding Knocki device IDs. If any entities are missing, please select "reload" from the device’s menu.
> 10. Go to settings in Home Assistant, and select "+ CREATE AUTOMATION" and then choose the "Create new automation" option.  
> 11. Select "+ADD TRIGGER."  

Reviewer: ```suggestion 
```

Author: @c0ffeeca7 I think all your changes make sense. Now wondering if this should be added in some place, I think it's important. Maybe, after login you'll see your Knocki tasks as entities or something like that. What do you think? Thanks!
> 9. You should see entities for each Knocki Task that you already created in the Knocki app, and the corresponding Knocki device IDs. If any entities are missing, please select "reload" from the device’s menu.

Reviewer 2: In an ideal world this would not be able to happen since the integration would update that. And I think that that logic is quite stable now right?

Author: @joostlek correct. Maybe just:
> You should see entities for each Knocki Task that you already created in the Knocki app, and the corresponding Knocki device IDs.

> a. Select "Entity" as the trigger type.  
> b. Search for the name of the task (as you named it in the Knocki app). Select the entity with that name appended to it.  
> c. Select "State" as the type of entity change.
> 12. Select "+ADD ACTION" and choose whatever action(s) you’d like the automation to trigger.

Reviewer: ```suggestion
```

> 13. Finally, test the action by tapping the associated gesture pattern. If the new automation isn’t triggering successfully, you need to select the "reload" option from the devices page for Knocki on Home Assistant, and it should start working.

Reviewer: ```suggestion
```

4. On the app’s device screen, select a gesture or [add a gesture](https://support.knocki.com/hc/en-us/articles/360013333634-Creating-a-Gesture).
5. Next, select the [add a task](https://support.knocki.com/hc/en-us/articles/12920956118291-How-do-I-Add-Tasks-to-a-Knocki) option.
6. Scroll down the task menu and select Home Assistant.
7. Follow the in-app instructions to add the Home Assistant task.

{% include integrations/config_flow.md %}

## Automation example

1. Go to {% my automations title="**Settings** > **Automations & scenes**" %} and in the lower right corner, select  **Create Automation**.
2. Select **Create new automation**.
3. Select **Add Trigger**, select **Entity** as the trigger type.
   - Search for the name of the task (as you named it in the Knocki app). Select the entity with that name appended to it.
   - Select **State** as the type of entity change.
4. Select **Add action** and choose whatever action(s) you’d like the automation to trigger.
5. Finally, test the action by tapping the associated gesture pattern. 

## Troubleshooting

> If the new automation isn’t triggering successfully, select the **reload** option from the devices page for Knocki on Home Assistant. It should then start working.

Author: @c0ffeeca7 I deleted troubleshooting section. Once this merged I'll create a new one with that section as we previously discussed.

> If the new automation isn’t triggering successfully, select the **reload** option from the devices page for Knocki on Home Assistant. It should then start working.

Reviewer: Thank you!

10. Go to settings in Home Assistant, and select "+ CREATE AUTOMATION" and then choose the "Create new automation" option.  
11. Select "+ADD TRIGGER."  
a. Select "Entity" as the trigger type.  
b. Search for the name of the task (as you named it in the Knocki app). Select the entity with that name appended to it.  
c. Select "State" as the type of entity change.
12. Select "+ADD ACTION" and choose whatever action(s) you’d like the automation to trigger.
13. Finally, test the action by tapping the associated gesture pattern. If the new automation isn’t triggering successfully, you need to select the "reload" option from the devices page for Knocki on Home Assistant, and it should start working.

## Troubleshooting

### My Knocki devices or entities in aren’t showing up (or aren’t updating) on Home Assistant

Knocki devices and entities will not be listed in Home Assistant until you’ve added a Home Assistant tasks via the Knocki mobile app. See steps 1-7. If they still aren’t showing up or aren’t updating, please select “reload” from the device’s menu in Home Assistant.

### There’s another issue I need help with

> If you’re experiencing any other issues or need general help, please reach out to [support@knocki.com](mailto:support@knocki.com).

Reviewer: Please remove the Troubleshooting section from this PR and create a separate PR for this.

Author: @c0ffeeca7 just fixed this. Thanks!

Author: @c0ffeeca7 do you know if we can embed a video using an Vimeo `iframe`?

Reviewer: Not Vimeo, no. But you could reference a video from YT using the following syntax:

`<lite-youtube videoid="XyBy0ckkiDU" videoStartAt="2047" videotitle="A Home-Approved Dashboard" posterquality="maxresdefault"></lite-youtube>`

Author: @c0ffeeca7 thanks. We're going to create a new PR then when we create and upload the video. Can we merge this now?

ha_integration_type: device
---

The **Knocki** {% term integration %} allows you to trigger your favorite automations simply by tapping custom patterns (such as triple taps) on ordinary surfaces.

## What is Knocki?

[Knocki](http://knocki.com) is a small device that transforms common surfaces (including walls, tables, countertops, furniture, doors, and more) into touch control interfaces. You can configure specific tap patterns (such as a triple-tap, or two double-taps) to trigger your favorite Home Assistant automations!

Unlike voice control, Knocki has no microphone, so it never listens in.  Plus, it allows your favorite functions to be within reach, without having to access a phone screen, button, or switch.

## Prerequisites

Make sure you have a [Knocki device](http://knocki.com) linked to your Knocki account before you set up this integration.
If you don’t have a Knocki account, it’s fast and easy to set one up using the free Knocki mobile app.

## Step-by-Step Instructions

Start the setup process in the knocki mobile app (steps 1-7), and complete the process in Home Assistant (steps 8-13). **Steps 8-10 cannot be completed prior to steps 1-7.**

> ### Steps 1-7:  Start the setup process in the Knocki Mobile App

Reviewer 3: This now mixes pre-requisites, setting up the integration, and an example an automation.

Please split those up and restore the use of our configuration flow include/template.

Author: Hey @frenck! Thanks for your feedback. 

> This now mixes pre-requisites, setting up the integration, and an example an automation.
What steps would you put in `Prerequisites`? Should we remove the `What is Knocki?` section?

Thanks!

Author: @frenck something like this maybe?
![image](https://github.com/user-attachments/assets/4aa30e77-6195-4b1b-8200-bac960bcd916)
